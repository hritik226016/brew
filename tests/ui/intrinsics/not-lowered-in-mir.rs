//! Check that intrinsics that do not get lowered to MIR, but are marked as such,
//! cause an error instead of silently invoking the body.
#![feature(intrinsics)]
//@ build-fail
//@ failure-status:101
//@ normalize-stderr-test: ".*note: .*\n\n" -> ""
//@ normalize-stderr-test: "thread 'rustc' panicked.*:\n.*\n" -> ""
//@ normalize-stderr-test: "internal compiler error:.*: " -> ""
//@ rustc-env:RUST_BACKTRACE=0

#[rustc_intrinsic]
#[rustc_intrinsic_lowers_to_mir]
pub const unsafe fn const_deallocate(_ptr: *mut u8, _size: usize, _align: usize) {}

fn main() {
    unsafe { const_deallocate(std::ptr::null_mut(), 0, 0) }
    //~^ ERROR: Intrinsic const_deallocate was marked as #[rustc_intrinsic_lowers_to_mir] but wasn't lowered by `LowerIntrinsics` pass
}
