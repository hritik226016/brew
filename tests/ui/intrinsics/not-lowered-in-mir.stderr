error: Intrinsic const_deallocate was marked as #[rustc_intrinsic_lowers_to_mir] but wasn't lowered by `LowerIntrinsics` pass
  --> $DIR/not-lowered-in-mir.rs:16:14
   |
LL |     unsafe { const_deallocate(std::ptr::null_mut(), 0, 0) }
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

query stack during panic:
#0 [mir_drops_elaborated_and_const_checked] elaborating drops for `main`
#1 [analysis] running analysis passes on this crate
end of query stack
error: aborting due to 1 previous error

